src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -20,12 +20,29 @@ internal static class SymbolExtensions
     private const string NamespaceMongoDBDriver = "MongoDB.Driver";
     private const string NamespaceMongoDBLinq = "MongoDB.Driver.Linq";
     private const string NamespaceSystemLinq = "System.Linq";
+    private const string AssemblyMongoDBBson = "MongoDB.Bson";
+    private const string NamespaceMongoDBBson = "MongoDB.Bson.Serialization.Attributes";
 
     private static readonly HashSet<string> s_supportedCollections = new()
     {
         "System.Collections.Generic.List<T>",
         "System.Collections.Generic.IList<T>",
-        "System.Collections.Generic.IEnumerable<T>"
+        "System.Collections.Generic.IEnumerable<T>",
+        "System.Collections.Generic.Dictionary<TKey, TValue>",
+        "System.Collections.Generic.IDictionary<TKey, TValue>"
+    };
+
+    private static readonly HashSet<string> s_supportedStructs = new()
+    {
+        "System.DateTime",
+        "System.TimeSpan"
+    };
+
+    private static readonly HashSet<string> s_unsupportedMongoAttributes = new()
```

Comment:
Lets do the opposite, support only what we support and have tests for.

Reply:
Done